src/sidebar/ConnectedSidebar.test.tsx

```diff
@@ -98,6 +107,45 @@ describe("SidebarApp", () => {
 
     await waitForEffect();
 
+    // The navigation listener should not be added for MV2
+    expect(
+      browser.webNavigation.onBeforeNavigate.addListener,
+    ).not.toHaveBeenCalled();
+
     expect(asFragment()).toMatchSnapshot();
   });
+
+  describe("mv3", () => {
+    beforeEach(() => {
+      jest.mocked(isMV3).mockReturnValue(true);
+    });
+
+    test("it registers the navigation listener", async () => {
+      await mockAuthenticatedMeApiResponse();
+      const { unmount } = render(
+        <MemoryRouter>
+          <ConnectedSidebar />
+        </MemoryRouter>,
+        {
+          setupRedux(dispatch) {
+            dispatch(authActions.setAuth(authStateFactory()));
+          },
+        },
+      );
+
+      await waitForEffect();
+
+      // The navigation listener should be added for MV3
+      expect(
+        browser.webNavigation.onBeforeNavigate.addListener,
+      ).toHaveBeenCalledWith(expect.any(Function));
```

Comment:
Does this have to use `waitForEffect()`? Could this be done instead as:
```suggestion
      await waitFor(() => {
         // The navigation listener should be added for MV3
        expect(
          browser.webNavigation.onBeforeNavigate.addListener,
        ).toHaveBeenCalledWith(expect.any(Function));
      });
```

Reply:
Turns out it's not really even needed